src/content/it/fundamentals/architecture/app-shell.md

```diff
@@ -0,0 +1,313 @@
+href="https://github.com/insin/react-hn">ReactHN</a> , <a
+href="https://github.com/GoogleChrome/sw-precache/tree/master/app-shell-demo">iFixit</a>).
+
+### Memorizzazione nella cache della shell dell'applicazione {: #app-shell-caching }
+
+Una app shell può essere memorizzata nella cache utilizzando un service worker
+scritto manualmente o un service worker generato utilizzando uno strumento di
+precaching di asset statici come
+[sw-precache](https://github.com/googlechrome/sw-precache).
+
+Note: gli esempi sono forniti solo a scopo informativo e illustrativo. Le
+risorse effettive utilizzate saranno probabilmente diverse per la tua
+applicazione.
+
+#### Memorizzazione manuale della app shell
+
+Di seguito è riportato un esempio di service worker che memorizza nella cache le
+risorse statiche dell'app shell nella [Cache
+API](https://developer.mozilla.org/en-US/docs/Web/API/Cache) utilizzando
+l'evento `install` del service worker:
+
+```
+var cacheName = 'shell-content';
+var filesToCache = [
+  '/css/styles.css',
+  '/js/scripts.js',
+  '/images/logo.svg',
+
+  '/offline.html’,
+
+  '/’,
+];
+
+self.addEventListener('install', function(e) {
+  console.log('[ServiceWorker] Install');
+  e.waitUntil(
+    caches.open(cacheName).then(function(cache) {
+      console.log('[ServiceWorker] Caching app shell');
+      return cache.addAll(filesToCache);
+    })
+  );
+});
+```
+
+#### Utilizza sw-precache per memorizzare nella cache l'app shell
+
+Il service worker generato da sw-precache memorizzerà nella cache e servirà le
+risorse configurate come parte del processo di generazione. Puoi avere in
+precache ogni file HTML, JavaScript e CSS che costituisce app shell. Tutto
+funzionerà offline e si caricherà velocemente nelle visite successive senza
+alcuno sforzo aggiuntivo.
+
+Ecco un esempio base dell'utilizzo di sw-precache come parte di un processo di
+generazione [gulp](http://gulpjs.com) :
+
+```
+gulp.task('generate-service-worker', function(callback) {
+  var path = require('path');
+  var swPrecache = require('sw-precache');
+  var rootDir = 'app';
+
+  swPrecache.write(path.join(rootDir, 'service-worker.js'), {
+    staticFileGlobs: [rootDir + '/**/*.{js,html,css,png,jpg,gif}'],
+    stripPrefix: rootDir
+  }, callback);
+});
+```
+
+Per ulteriori informazioni sulla memorizzazione nella cache delle risorse
+statiche consultare il codelab [Adding a Service Worker with
+sw-precache](https://codelabs.developers.google.com/codelabs/sw-precache/index.html?index=..%2F..%2Findex#0).
+
+Note: sw-precache è utile per il caching offline delle risorse statiche. Per le
+risorse runtime/dinamiche, ti consigliamo di utilizzare la nostra libreria
+gratuita [sw-toolbox](https://github.com/googlechrome/sw-toolbox) .
+
+## Conclusione {: #conclusion }
+
+Una shell di app che utilizza Service worker è un modello efficace per il
+caching offline, ma offre anche vantaggi significativi in ​​termini di
```

Comment:
memorizzazione della cache offline e offre anche vantaggi significativi in ​​termini di